sympy/relation/binrel.py

```diff
@@ -0,0 +1,295 @@
+        elif r.lhs.is_number:
+            r = r.reversed
+        elif tuple(ordered(args)) != args:
+            r = r.reversed
+
+        LHS_CEMS = getattr(r.lhs, 'could_extract_minus_sign', None)
+        RHS_CEMS = getattr(r.rhs, 'could_extract_minus_sign', None)
+
+        if isinstance(r.lhs, BooleanAtom) or isinstance(r.rhs, BooleanAtom):
+            return r
+
+        # Check if first value has negative sign
+        if LHS_CEMS and LHS_CEMS():
+            return r.reversedsign
+        elif not r.rhs.is_number and RHS_CEMS and RHS_CEMS():
+            # Right hand side has a minus, but not lhs.
+            # How does the expression with reversed signs behave?
+            # This is so that expressions of the type
+            # Q.eq(x, -y) and Q.eq(-x, y)
+            # have the same canonical representation
+            expr1, _ = ordered([r.lhs, -r.rhs])
+            if expr1 != r.lhs:
+                return r.reversed.reversedsign
+
+        return r
+
+    @property
+    def binary_symbols(self):
+        return set()
+
+    def as_Relational(self):
+        return self.function.as_Relational(*self.arguments)
+
+    def _eval_ask(self, assumptions):
+        # quick exit for structurally same arguments
+        ret = self.function._compare_reflexive(self.lhs, self.rhs)
+        if ret is not None:
+            return ret
+
+        lhs, rhs = self.lhs.refine(assumptions), self.rhs.refine(assumptions)
+        r = self.function(lhs, rhs).simplify()
+
+        # attempt quick exit again with simplified arguments
+        ret = r.function._compare_reflexive(r.lhs, r.rhs)
+        if ret is not None:
+            return ret
+
+        # use multipledispatch handler
+        return r.function.eval(r.arguments, assumptions)
+
+    def simplify(self, equation=True, side="all", **kwargs):
+        """
+        Simplify *self* without evaluating to boolean value. Assumption
+        is not taken into consideration.
+
+        Parameters
+        ==========
+
+        equation : bool, optional
+            If ``True``, simplify both sides and canonical result.
+            If ``False``, each side is simplified but not canonicalized.
+            You can decide which side to be simplified by *side* argument.
+
+        side : "all", "lhs" or "rhs", optional
+            Specify which side the rewriting will be done. Only valid
+            when *equation* is ``True``. Default is "all".
+
+        Examples
+        ========
+
+        >>> from sympy import cos, gamma, sin, Q
+        >>> from sympy.abc import x
+        >>> eqn = Q.eq(sin(x)**2 + cos(x)**2, gamma(x)/gamma(x-2))
+        >>> eqn.simplify()
+        x**2 - 3*x = -1
+        >>> eqn.simplify(equation=False)
+        1 = (x - 2)*(x - 1)
+        >>> eqn.simplify(equation=False, side='lhs')
+        1 = gamma(x)/gamma(x - 2)
+        """
+        kwargs.update(equation=equation,
+                      side=side)
+        return simplify(self, **kwargs)
```

Comment:
The defaults in `_eval_simplify` are the same as here so this method is only needed for the docstring.

Would it maybe be better to have specific methods like `simplifylhs`?

Reply:
Having `simplify`, `simplifylhs`, and `simplifyrhs` seems redundant. That design implies that we should have `refinelhs`, `rewriterhs`, `doitlhs` and myriads of more.

Reply:
I defined this method only for docstring indeed. Shall I make class docstring elaborate on `.simplify()` and delete this method?

Reply:
I think that controlling this with a custom argument to simplify like `simplify(obj, side='left')` just seems like a strange API to me. I can't imagine wanting to do that with some object that has embedded equalities and if I'm working with an equation directly it would seem more natural to have a more direct way of operating in the different sides (for any function, not just simplify).

Somewhere else for `Equation` I proposed:
```
eq.dolhs.simplify()
eq.dorhs.simplify()
eq.do.simplify()
```
where `dolhs` etc are proxy objects.

Reply:
See the working implementation in #19479, starting at line 357 of equation.py
```
    class _sides:
        def __init__(self,eqn, side='both'):
            self.eqn = eqn
            self.side = side

        def __getattr__(self, name):
            func = None
            if self.side in ('rhs', 'both'):
                func = getattr(self.eqn.rhs, name, None)
            else:
                func = getattr(self.eqn.lhs, name, None)
            return functools.partial(self.eqn.apply, func, side=self.side)

    @property
    def do(self):
        return self._sides(self, side='both')

    @property
    def dolhs(self):
        return self._sides(self, side='lhs')

    @property
    def dorhs(self):
        return self._sides(self, side='rhs')
```

Reply:
Proxy object is a clever idea. I will try.

Reply:
I implemented this. Please take a look again!

Reply:
I removed proxy object from this PR for now in order to limit the scope into basic implementation. It will be done in succeeding PRs.